sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -47,20 +49,39 @@ object ResolveHints {
    *
    * This rule must happen before common table expressions.
    */
-  class ResolveBroadcastHints(conf: SQLConf) extends Rule[LogicalPlan] {
+  class ResolveBroadcastHints(conf: SQLConf, catalog: SessionCatalog) extends Rule[LogicalPlan] {
     private val BROADCAST_HINT_NAMES = Set("BROADCAST", "BROADCASTJOIN", "MAPJOIN")
 
     def resolver: Resolver = conf.resolver
 
-    private def applyBroadcastHint(plan: LogicalPlan, toBroadcast: Set[String]): LogicalPlan = {
+    private def namePartsWithDatabase(nameParts: Seq[String]): Seq[String] = {
+      if (nameParts.size == 1) {
+        catalog.getCurrentDatabase +: nameParts
+      } else {
+        nameParts
+      }
+    }
+
+    private def matchedTableIdentifier(
+        nameParts: Seq[String],
+        tableIdent: IdentifierWithDatabase): Boolean = {
+      val identifierList =
+        tableIdent.database.getOrElse(catalog.getCurrentDatabase) :: tableIdent.identifier :: Nil
+      namePartsWithDatabase(nameParts).corresponds(identifierList)(resolver)
```

Comment:
This logic will make a regression (`plan1` in the below) in case of global temporary view. Please add the following test case into `GlobalTempViewSuite` and revise the logic to handle both cases correctly.

```scala
test("broadcast hint on global temp view") {
    import org.apache.spark.sql.catalyst.plans.logical.{ResolvedHint, Join}

    spark.range(10).createGlobalTempView("v1")
    spark.range(10).createOrReplaceTempView("v2")

    val plan1 = sql(s"SELECT /*+ BROADCAST(v1) */ * FROM global_temp.v1, v2 WHERE v1.id = v2.id")
      .queryExecution.optimizedPlan
    assert(plan1.asInstanceOf[Join].left.isInstanceOf[ResolvedHint])
    assert(!plan1.asInstanceOf[Join].right.isInstanceOf[ResolvedHint])

    val plan2 = sql(
      s"SELECT /*+ BROADCAST(global_temp.v1) */ * FROM global_temp.v1, v2 WHERE v1.id = v2.id")
      .queryExecution.optimizedPlan
    assert(plan2.asInstanceOf[Join].left.isInstanceOf[ResolvedHint])
    assert(!plan2.asInstanceOf[Join].right.isInstanceOf[ResolvedHint])
  }
```

Reply:
@dongjoon-hyun a little confused about the name resolution here;
```
"SELECT /*+ MAPJOIN(v1) */ * FROM global_temp.v1, v2 WHERE v1.id = v2.id",
```
`MAPJOIN(v1)` implicitly means `global_temp.v1`?
For example;
```
"SELECT /*+ MAPJOIN(v1) */ * FROM default.v1, global_temp.v1 WHERE default.v1.id = global_temp.v1.id",
```
In this case, what's the `MAPJOIN(v1)` behaviour?
 - 1. Apply no hint
 - 2. Apply a hint into `default.v1` only
 - 3. Apply a hint into both

WDYT?

Reply:
BTW, @maropu . In addition,
- The current behavior of `master` branch (Spark 2.4) is `Apply a hint into both`.
- The legacy behavior of Spark 2.3.1 is raising an exception for that query.

So, I think it's a good change to become consistent in Spark 2.4.

Reply:
oh, yes. I'll refine the pr. thanks.